src/database.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the DatabaseObserver class to handle database relation and state."""
+
+import typing
+
+from charms.data_platform_libs.v0.data_interfaces import DatabaseRequires
+from ops.charm import CharmBase
+from ops.framework import Object
+
+from constants import DATABASE_NAME
+
+class DatabaseService(Object):
+    """The Database relation observer."""
+
+    def __init__(self, charm: CharmBase, relation_name: str):
+        """Initialize the oserver and register event handlers.
+
+        Args:
+            charm: The parent charm to attach the observer to.
+        """
+        super().__init__(charm, "database-observer")
```

Comment:
Service or Observer? What's the difference?

Reply:
None, at least to me :) I saw the mention of Service in the discourse-k8s charm, though it should be an Observer.